execute/window_test.go

```diff
@@ -509,3 +510,270 @@ func errAsString(err error) (s string) {
+			t: execute.Time(6 * time.Minute),
+			want: testBounds{
+				Start: execute.Time(5 * time.Minute),
+				Stop:  execute.Time(10 * time.Minute),
+			},
+		},
+		{
+			name: "simple with negative period",
+			w: mustWindowInterval(
+				values.ConvertDurationNsecs(5*time.Minute),
+				values.ConvertDurationNsecs(-5*time.Minute),
+				values.ConvertDurationNsecs(30*time.Second)),
+			t: execute.Time(5 * time.Minute),
+			want: testBounds{
+				Start: execute.Time(30 * time.Second),
+				Stop:  execute.Time(5*time.Minute + 30*time.Second),
+			},
+		},
+		{
+			name: "simple with offset",
+			w: mustWindowInterval(
+				values.ConvertDurationNsecs(5*time.Minute),
+				values.ConvertDurationNsecs(5*time.Minute),
+				values.ConvertDurationNsecs(30*time.Second)),
+			t: execute.Time(5 * time.Minute),
+			want: testBounds{
+				Start: execute.Time(30 * time.Second),
+				Stop:  execute.Time(5*time.Minute + 30*time.Second),
+			},
+		},
+		{
+			name: "simple with negative offset",
+			w: mustWindowInterval(
+				values.ConvertDurationNsecs(5*time.Minute),
+				values.ConvertDurationNsecs(5*time.Minute),
+				values.ConvertDurationNsecs(-30*time.Second)),
+			t: execute.Time(5 * time.Minute),
+			want: testBounds{
+				Start: execute.Time(4*time.Minute + 30*time.Second),
+				Stop:  execute.Time(9*time.Minute + 30*time.Second),
+			},
+		},
+		{
+			name: "simple with equal offset before",
+			w: mustWindowInterval(
+				values.ConvertDurationNsecs(5*time.Minute),
+				values.ConvertDurationNsecs(5*time.Minute),
+				values.ConvertDurationNsecs(5*time.Minute)),
+			t: execute.Time(0),
+			want: testBounds{
+				Start: execute.Time(0 * time.Minute),
+				Stop:  execute.Time(5 * time.Minute),
+			},
+		},
+		{
+			name: "simple with equal offset after",
+			w: mustWindowInterval(
+				values.ConvertDurationNsecs(5*time.Minute),
+				values.ConvertDurationNsecs(5*time.Minute),
+				values.ConvertDurationNsecs(5*time.Minute)),
+			t: execute.Time(7 * time.Minute),
+			want: testBounds{
+				Start: execute.Time(5 * time.Minute),
+				Stop:  execute.Time(10 * time.Minute),
+			},
+		},
+		{
+			name: "simple months",
+			w: mustWindowInterval(
+				values.ConvertDurationMonths(5),
+				values.ConvertDurationMonths(5),
+				values.ConvertDurationMonths(0)),
+			t: mustTime("1970-01-01T00:00:00Z"),
+			want: testBounds{
+				Start: mustTime("1970-01-01T00:00:00Z"),
+				Stop:  mustTime("1970-06-01T00:00:00Z"),
+			},
+		},
+		{
+			name: "simple months with offset",
+			w: mustWindowInterval(
+				values.ConvertDurationMonths(3),
+				values.ConvertDurationMonths(3),
+				values.ConvertDurationMonths(1)),
+			t: mustTime("1970-01-01T00:00:00Z"),
+			want: testBounds{
+				Start: mustTime("1969-11-01T00:00:00Z"),
+				Stop:  mustTime("1970-02-01T00:00:00Z"),
+			},
+		},
+	}
+	for i := 0; i < 100; i++ {
+		for _, tc := range testcases {
+			tc := tc
+			b.Run(tc.name, func(b *testing.B) {
+				got := tc.w.GetLatestBounds(tc.t)
```

Comment:
Benchmarks need to be of the form:

```
for i := 0; i < b.N; i++ {
    // code to benchmark
}
```

For future reference, it's also not really necessary to check if the values you got were good or not. Test cases are used for that. Benchmarks are mostly used to verify that things go fast. For this case, I think these tests are good since it prevents the optimizer from just removing the entire block and I don't think it will impact the benchmark result.